geo.py

```diff
@@ -0,0 +1,247 @@
+            # Pick first matching placename, w/o disambiguating
+            # TODO: This could be smarter
+            coords = coords_from_addr_info(info[0])
+
+    if coords:
+        (loc["latitude"], loc["longitude"]) = coords
+
+    return loc
+
+
+def coords_for_country(iso_code):
+    """ Return coordinates for a given country code """
+    assert len(iso_code) == 2
+
+    iso_code = iso_code.upper()
+
+    # Lazy-load data, save as function attribute
+    if not hasattr(coords_for_country, "iso2coord"):
+        jsonstr = resource_stream(__name__, COUNTRY_COORDS_JSONPATH).read().decode()
+        coords_for_country.iso2coord = json.loads(jsonstr)
+
+    return coords_for_country.iso2coord.get(iso_code)
+
+
+def coords_for_street_name(street_name, placename=None, placename_hints=[]):
+    """ Return coordinates for an Icelandic street name as a tuple. As some
+        street names exist in more than one place, we try to narrow it down 
+        to a single street if possible. Street coordinates are the coordinates
+        of the lowest house number. """
+
+    addresses = iceaddr_lookup(street_name, placename=placename, limit=100)
+
+    if not len(addresses):
+        return None
+
+    # Find all places containing street_name
+    places = set(a.get("stadur_nf") for a in addresses)
+    addr = None
+
+    # Only exists in one place
+    if len(places) == 1:
+        addr = addresses[0]
+    elif placename_hints:
+        # See if placename hints can narrow it down
+        for pn in placename_hints:
+            addresses = iceaddr_lookup(street_name, placename=pn)
+            places = set(a.get("stadur_nf") for a in addresses)
+            if len(places) == 1:
+                addr = addresses[0]
+                break
+
+    return coords_from_addr_info(addr)
+
+
+def coords_from_addr_info(info):
+    """ Get coordinates from the address dict provided by iceaddr """
+    if info and info.get("lat_wgs84") and info.get("long_wgs84"):
+        return (info["lat_wgs84"], info["long_wgs84"])
+    return None
+
+
+def country_name_for_isocode(iso_code, lang=ICELANDIC_LANG_ISOCODE):
+    """ Return country name for an ISO 3166-1 alpha-2 code """
+    assert len(iso_code) == 2
+    assert len(lang) == 2
+
+    iso_code = iso_code.upper()
+    lang = lang.lower()
+
+    if not lang in available_languages():
+        return None
+
+    countries = dict(countries_for_language(lang))
+    return countries.get(iso_code)
+
+
+def isocode_for_country_name(country_name, lang=ICELANDIC_LANG_ISOCODE):
+    """ Return the ISO 3166-1 alpha-2 code for a country 
+        name in the specified language (two-char ISO 639-1) """
+    assert len(lang) == 2
+
+    lang = lang.lower()
+    if not lang in available_languages():
+        return None
+
+    countries = countries_for_language(lang)
+    for iso_code, name in countries:
+        if name == country_name:
+            return iso_code
+
+    additions = {
```

Comment:
Put this constant at module scope, at top of file?

Reply:
...or, even better going forward, in a `.conf` file that is read in `settings.py`.